File: 7.Sorting/SelectionSort/selection_sort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

void SelectionSort(int arr[], int n)
{
    int i, j, pos, least;
    for (i = 0; i < n; i++)
    {
        least = arr[i];
        pos = i;
        for (j = i + 1; j < n; j++)
        {
            if (arr[j] < least)
            {
                least = arr[j];
                pos = j;
            }
        }
        if (i != pos)
        {
            swap(&arr[i], &arr[pos]);
        }
    }
}

void display(int arr[], int size)
{
    int i;
    for (i = 0; i < size; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
}


int main()
{
    int A[100000], n, i;
    srand(time(NULL));
    clock_t start, end, diff;
    printf("Enter n:");
    scanf("%d", &n);
    for (i = 0; i < n; i++)
    {
        A[i] = rand();
    }
    display(A, n);
    start = clock();
    SelectionSort(A, n);
    end = clock();
    display(A, n);
    diff = end - start;
    printf("The time taken is %f Seconds\n", (double)diff / CLOCKS_PER_SEC);
    return 0;
}

File: 7.Sorting/SelectionSort/selection_sort.py
import time

def selectionSort(array):
    for i in range(len(array)):
        minIndex = i
        for j in range(i + 1, len(array)):
            if array[j] < array[minIndex]:
                minIndex = j
        array[i], array[minIndex] = array[minIndex], array[i]
    return array

time1 = time.time()
arr = [64, 34, 25, 12, 22, 11, 90]
print(selectionSort(arr))
time2 = time.time()
print(f"Time taken to execute : {time2 - time1}")